File: app/src/main/java/mk/webfactory/template/network/api/ServiceModule.java
package mk.webfactory.template.network.api;

import dagger.Module;
import dagger.Provides;
import javax.inject.Singleton;
import retrofit2.Retrofit;

@Module public class ServiceModule {

  @Provides @Singleton public UserService provideUserService(Retrofit retrofit) {
    return retrofit.create(UserService.class);
  }
}

File: app/src/main/java/mk/webfactory/template/network/api/ServiceModule.kt
package mk.webfactory.template.network.api

import dagger.Module
import dagger.Provides
import retrofit2.Retrofit
import javax.inject.Singleton

@Module
class ServiceModule {
    @Provides
    @Singleton
    fun provideUserService(retrofit: Retrofit): UserService {
        return retrofit.create(UserService::class.java)
    }
}